vms/administrator/utils.py

```diff
@@ -14,3 +16,23 @@ def wrapped_view(request, *args, **kwargs):
         return func(request, *args, **kwargs)
 
     return wrapped_view
+
+
+def admin_id_check(func):
+    @wraps(func)
+    def wrapped_view(request, admin_id):
+        administrator = getattr(request.user, 'administrator',
+                      hasattr(request.user, 'administrator'))
+        if administrator is None:
+            return render(request, 'vms/no_admin_rights.html', status=403)
+        elif administrator is False:
+            admin = Administrator.objects.get(id=admin_id)
+            if not admin:
```

Comment:
this will never happen. If an admin cannot be found on a `get`, it would raise an Administrator.DoesNotExist

Reply:
Ok. So i will add a try except then? @tonythomas01

Reply:
no. Line 30 will not be executed by any chance. The whole function needs to be improved programmatically.